### PR TITLE
preserve data

### PR DESCRIPTION
**Pull request by Abdulwhbhussain**
# Project - 6: [OOP school library: preserve data](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_preserve_data.md#oop-school-library-preserve-data)

### Implemented Tasks:

* Check the documentation for Ruby's [JSON library](https://ruby-doc.org/stdlib-3.0.0/libdoc/json/rdoc/JSON.html).
* On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. rentals.json) 
* created `rentals.json, books.json, and people.json`.
* created `save_data` method that stores the newly added data to the JSON files using JSON methods `write` and `generate`.
* On startup of your app load all data from files. using the `load_data` method.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -86,13 +88,63 @@ def rentals_of_person(id)
     end
   end
 
+  def save_data()
+    File.write('books.json', JSON.pretty_generate(@books))
+    File.write('people.json', JSON.pretty_generate(@people))
+    File.write('rentals.json', JSON.pretty_generate(@rentals))
```
Comment (shahadat3669) [inline app.rb:94]: - [x] Well! It appears that your `books.json` file is correctly saving the data. However, there seems to be an issue with the `people.json` and `rentals.json` files, as they are not saving the data correctly. As you can see in the screenshot below, the `name` values are being saved in the `age` field, and the `age` values are being saved in the `name` field. Please update your code to ensure that data is correctly saved in these files. Thank you for your attention to detail, and keep up the good work! 👍📚🔍
image
- [x] I also noticed that if I don't add a new people or rental and run the app again and exit, your files change and add proper data properly. But they should be done correctly in the first place without running the app again. Additionally, if I add a new people, some people have the right data, while others have wrong data. Please consider updating your code to address these issues.
image

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @Abdulwhbhussain, @tan12082001, 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ On exit from the student's app check the `book` data is being saved
✅ Saving each `book`, `person`, and `rental` data in a separate file.
✅ On startup of their app load all data from files


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

 

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (Abdulwhbhussain) [reply app.rb:94]: Fixed This Bug!

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @Abdulwhbhussain, and @tan12082001👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Good job with preserving data. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**